### PR TITLE
Document Image Processing example

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR adds a simple Python script and a mkdocs file to document the ttk-data ImageProcessing state file.

Enjoy,
Pierre

### CODE REVIEW

**Comment by julien-tierny** [comment]
once your PR is merged, thanks for including a link to this example in the doxygen and paraview documentation of the involved modules. see this commit for instance:
topology-tool-kit/ttk@2b9fd4c

File: python/imageProcessing.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+naturalImagepng = PNGSeriesReader(FileNames=["naturalImage.png"])
+
+# create a new 'Compute Derivatives'
+computeDerivatives1 = ComputeDerivatives(Input=naturalImagepng)
+computeDerivatives1.Scalars = ["POINTS", "PNGImage"]
+computeDerivatives1.Vectors = [None, ""]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

File: python/imageProcessing.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+# create a new 'Calculator'
+calculator1 = Calculator(Input=cellDatatoPointData1)
+calculator1.ResultArrayName = "gradient"
+calculator1.Function = "mag(ScalarGradient)"
+
+# create a new 'TTK PersistenceDiagram'
+tTKPersistenceDiagram1 = TTKPersistenceDiagram(Input=calculator1)
+tTKPersistenceDiagram1.ScalarField = ["POINTS", "gradient"]
+tTKPersistenceDiagram1.InputOffsetField = [None, ""]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

File: python/imageProcessing.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+# create a new 'Threshold'
+threshold1 = Threshold(Input=tTKPersistenceDiagram1)
+threshold1.Scalars = ["CELLS", "PairIdentifier"]
+threshold1.ThresholdRange = [-0.1, 59608.0]
+
+# create a new 'Extract Surface'
+extractSurface2 = ExtractSurface(Input=threshold1)
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove.

File: python/imageProcessing.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+# create a new 'Threshold'
+persistenceThreshold = Threshold(Input=threshold1)
+persistenceThreshold.Scalars = ["CELLS", "Persistence"]
+persistenceThreshold.ThresholdRange = [6.0, 9999.0]
+
+# create a new 'Extract Surface'
+extractSurface1 = ExtractSurface(Input=persistenceThreshold)
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=extractSurface1)
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove.

File: python/imageProcessing.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+# create a new 'Extract Surface'
+extractSurface2 = ExtractSurface(Input=threshold1)
+
+# create a new 'Threshold'
+persistenceThreshold = Threshold(Input=threshold1)
+persistenceThreshold.Scalars = ["CELLS", "Persistence"]
+persistenceThreshold.ThresholdRange = [6.0, 9999.0]
+
+# create a new 'Extract Surface'
+extractSurface1 = ExtractSurface(Input=persistenceThreshold)
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove.

File: python/imageProcessing.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=extractSurface1)
+tTKIcospheresFromPoints1.Radius = 0.65
+
+# create a new 'TTK TopologicalSimplification'
+tTKTopologicalSimplification1 = TTKTopologicalSimplification(
+    Domain=calculator1,
+    Constraints=persistenceThreshold,
+)
+tTKTopologicalSimplification1.ScalarField = ["POINTS", "gradient"]
+tTKTopologicalSimplification1.InputOffsetField = [None, ""]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

File: python/imageProcessing.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+tTKTopologicalSimplification1.VertexIdentifierField = [None, ""]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

File: python/imageProcessing.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex1 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification1)
+tTKMorseSmaleComplex1.ScalarField = ["POINTS", "gradient"]
+tTKMorseSmaleComplex1.OffsetField = ["POINTS", "OutputOffsetScalarField"]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

File: python/imageProcessing.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+# create a new 'Threshold'
+threshold2 = Threshold(Input=tTKPersistenceDiagram1)
+threshold2.Scalars = ["CELLS", "PairIdentifier"]
+threshold2.ThresholdRange = [-1.0, -0.1]
+
+# create a new 'Extract Surface'
+extractSurface3 = ExtractSurface(Input=threshold2)
+
+# create a new 'Tube'
+tube1 = Tube(Input=extractSurface3)
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove.

File: python/imageProcessing.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+)
+tTKIdentifierRandomizer1.ScalarField = ["POINTS", "DescendingManifold"]
+
+# create a new 'Threshold'
+threshold2 = Threshold(Input=tTKPersistenceDiagram1)
+threshold2.Scalars = ["CELLS", "PairIdentifier"]
+threshold2.ThresholdRange = [-1.0, -0.1]
+
+# create a new 'Extract Surface'
+extractSurface3 = ExtractSurface(Input=threshold2)
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove.

File: python/imageProcessing.py [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+# create a new 'TTK IdentifierRandomizer'
+tTKIdentifierRandomizer1 = TTKIdentifierRandomizer(
+    Input=OutputPort(tTKMorseSmaleComplex1, 3),
+)
+tTKIdentifierRandomizer1.ScalarField = ["POINTS", "DescendingManifold"]
+
+# create a new 'Threshold'
+threshold2 = Threshold(Input=tTKPersistenceDiagram1)
+threshold2.Scalars = ["CELLS", "PairIdentifier"]
+threshold2.ThresholdRange = [-1.0, -0.1]
```
Comment (julien-tierny) [inline python/imageProcessing.py]: dead leaf, please remove

File: docs/imageProcessing.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
+# Image Processing example
+
+![Image Processing example
+Image](https://topology-tool-kit.github.io/img/gallery/imageProcessing.jpg)
+
+
+## Pipeline description
+
+This example processes a black and white image (top left view on the
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: black and white -› greyscale

File: docs/imageProcessing.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+above screenshot) to generate a segmentation. We will construct the
+segmentation from the image gradient.
+
+First, the image is loaded from disk. The gradient is computed with
+ParaView's ComputeDerivatives or Gradient filters. Since TTK only
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: ComputeDerivatives -› `ComputeDerivatives`

File: docs/imageProcessing.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+works on scalar field, a Calculator is used to compute the gradient
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: Calculator -› `Calculator`

File: docs/imageProcessing.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+From the gradient magnitude, a simplification step involving
+[PersistenceDiagram](https://topology-tool-kit.github.io/doc/html/classttkPersistenceDiagram.html)
+(bottom left view) and
+[TopologicalSimplification](https://topology-tool-kit.github.io/doc/html/classttkTopologicalSimplification.html)
+helps removing the noise in the gradient (top right view).
+
+To segment the image, we use the
+[MorseSmaleComplex](https://topology-tool-kit.github.io/doc/html/classttkMorseSmaleComplex.html)
+filter. Since on the input image the objects correspond to low values
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: on the input image -› in the input image

File: docs/imageProcessing.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+in the gradient and their edges to high values, we are interested in
+the DescendingManifold scalar field of the Segmentation output, whose
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: DescendingManifold -› `DescendingManifold`

Comment (julien-tierny) [reply docs/imageProcessing.md]: Segmentation -› `Segmentation`

File: docs/imageProcessing.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+cells represent regions of low scalar field values.
+
+The
+[IdentifierRandomizer](https://topology-tool-kit.github.io/doc/html/classttkIdentifierRandomizer.html)
+filter is eventually used to ensure that neighboring cells are colored
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: to ensure (...) -› in order to color neighbor cells with a distinct color
(there is no guarantee)

File: docs/imageProcessing.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+## Python code
+
+``` python  linenums="1"
+--8<-- "python/imageProcessing.py"
+```
+
+## Inputs
+- [naturalImage.png](https://github.com/topology-tool-kit/ttk-data/raw/dev/naturalImage.png):
+  a black and white PNG image.
```
Comment (julien-tierny) [inline docs/imageProcessing.md]: a greyscale PNG picture.

**Review by julien-tierny — CHANGES_REQUESTED** [review]
thanks!
thanks for addressing the few remarks below.

**Comment by pierre-guillou** [comment]
Everything should be fixed/removed now

**Review by julien-tierny — APPROVED** [review]

**Comment by julien-tierny** [comment]
awesome, thanks!

**Pull request merged**